### PR TITLE
feat: Implement completions for named tuple fields

### PR DESCRIPTION
**Pull request by KacperFKorban**
closes scala#20478

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/interactive/Completion.scala [hunk 1]
```diff
@@ -516,6 +527,27 @@ object Completion:
           .toSeq
           .groupByName
 
+    /** Completions for named tuples */
+    private def namedTupleCompletions(qual: tpd.Tree)(using Context): CompletionMap =
+      def namedTupleCompleionsFromType(tpe: Type): CompletionMap =
+        tpe.namedTupleElementTypes
```
Comment (rochala) [inline compiler/src/dotty/tools/dotc/interactive/Completion.scala:533]: Shouldn't we create a new fresh context for this ?

File: compiler/src/dotty/tools/dotc/interactive/Completion.scala [hunk 2]
```diff
@@ -516,6 +527,27 @@ object Completion:
           .toSeq
           .groupByName
 
+    /** Completions for named tuples */
+    private def namedTupleCompletions(qual: tpd.Tree)(using Context): CompletionMap =
+      def namedTupleCompleionsFromType(tpe: Type): CompletionMap =
```
Comment (rochala) [inline compiler/src/dotty/tools/dotc/interactive/Completion.scala:532]:
```suggestion
      def namedTupleCompletionsFromType(tpe: Type): CompletionMap =
```

**Review by rochala — COMMENTED** [review]
Also, could you add one more test case in `scala3-language-server/CompletionTest.scala` ? Right now this is the place where we test only the compiler completions. In the future, it should be migrated to presentation compiler, but as of now this is a technical debt

File: compiler/src/dotty/tools/dotc/interactive/Completion.scala (hunk 1 above)
Comment (KacperFKorban) [reply compiler/src/dotty/tools/dotc/interactive/Completion.scala:533]: Probably, it will definitely be safer to do so. Good point.

**Review by rochala — APPROVED** [review]

**Pull request merged**